### ISSUE TITLE
MTL templates without enabling string templates?

### ISSUE THREAD

**Issue by mtdowling** [manifold-systems/manifold#124]
**Is your feature request related to a problem? Please describe.**

I want to use Manifold templates, but it pulls in a dependency on [manifold-strings](https://github.com/manifold-systems/manifold/blob/master/manifold-deps-parent/manifold-templates/pom.xml#L21), which seems to automatically enable string interpolation in Java source files. All I'm looking for is a templating language, and MTL looks amazing. At this point, I'm not interested in adding string interpolation to all of my .java files.

I'm building a framework that will be extended by other teams, and I don't want to ask each of these teams to either decorate all of their classes with `manifold.strings.api.DisableStringLiteralTemplates`, escape `$` found in every string, or implement the `manifold.strings.api.ITemplateProcessorGate` SPI.

I don't know if trusting all users of a framework to properly implement `ITemplateProcessorGate` is going to scale either. To do it properly, you'd need to only ensure that you exclude files that are part of your own package rather than blanket excluding files from other packages you don't own that might actually rely on string interpolation.

**Describe the solution you'd like**

Ideally, I could use manifold-templates without needing to enable string interpolation in all of my source files. Maybe there's a compiler setting to disable string interpolation except for when used in templates? Or maybe, you should have to explicitly enable string interpolation rather than disable it.

**Describe alternatives you've considered**

My idea for a workaround is to add an implementation of `ITemplateProcessorGate` that only allows interpolation on type names ending with "Template". I don't think this is going to work 5 years from now if Manifold becomes mainstream and people start relying on string interpolation.

```java
package com.foo.example;

import manifold.strings.api.ITemplateProcessorGate;

public class TemplateProcessorGate implements ITemplateProcessorGate {
    @Override
    public boolean exclude(String typeName) {
        return !typeName.endsWith("Template");
    }
}
```

**Additional context**

**Comment by rsmckinney** [comment]
@mtdowling Hi.  Makes good sense.  I tend to use string interpolation with `manifold-strings` in most projects now, but I can easily fall back to string concat and remove the dependency from `manifold-templates`. I'll include this change in the next release coming soon ~(7 - 10 days).  Thanks for reporting this!

**Comment by rsmckinney** [comment]
@mtdowling Update. The change for this did not make it v2019.1.21.  Now scheduled for release in v2019.1.23 within 10 days. Apologies for the delay.

**Comment by rsmckinney** [comment]
@mtdowling changes available now in release `2019.1.23`

**Comment by mtdowling** [comment]
Thanks!

**Comment by mtdowling** [comment]
I'm trying this out, and I'm having trouble with it. It looks like manifold-templates still has a dependency on manifold-strings: https://github.com/manifold-systems/manifold/blob/master/manifold-deps-parent/manifold-templates/pom.xml#L24-L28. This causes my project to fail to run if I am already using "$" inside of strings. This commit added the dependency back to manifold-templates after it was previously removed: https://github.com/manifold-systems/manifold/issues/181.

Is there something else I need to do to remove string interpolation?

**Comment by rsmckinney** [comment]
@mtdowling `manifold-templates` will change `manifold-strings` to an *optional* dependency, which will hide it from your project. This change will be released later this week. Thanks for reporting this!

**Comment by rsmckinney** [comment]
Fix released in version 2020.1.25

**Comment by mtdowling** [comment]
Thanks!
